src/FixedDataTable.js

```diff
@@ -957,12 +957,44 @@ var FixedDataTable = createReactClass({
       children.push(child);
     });
 
-    // Allow room for the scrollbar, less 1px for the last column's border
-    var adjustedWidth = props.width - Scrollbar.SIZE - Scrollbar.OFFSET;
+    var columns;
+    var columnGroups;
 
-    var useGroupHeader = false;
-    if (children.length && children[0].type.__TableColumnGroup__) {
-      useGroupHeader = true;
+    var tableWidth = props.width;
+    var useGroupHeader = children.length > 0 && children[0].type.__TableColumnGroup__ === true;
+
+    if (useGroupHeader) {
+      var columnGroupSettings =
+        FixedDataTableWidthHelper.adjustColumnGroupWidths(
+          children,
+          tableWidth
+      );
+      columns = columnGroupSettings.columns;
+      columnGroups = columnGroupSettings.columnGroups;
+    } else {
+      columns = FixedDataTableWidthHelper.adjustColumnWidths(
+        children,
+        tableWidth
+      );
+    }
+
+    var scrollContentWidth = FixedDataTableWidthHelper.getTotalWidth(columns);
+    var horizontalScrollbarVisible = scrollContentWidth > tableWidth &&
```

Comment:
Now tableWidth may change after this line is called.  It could lead to a case where we the we don't need a horizontal scrollbar, but then adding in the vertical scrollbar causes us to need one.

That's why this problem is hard... you can't decide on the scrollbar's visibility independently, you have to handle them together.